adam-core/src/test/scala/org/bdgenomics/adam/rdd/read/AlignmentRecordRDDSuite.scala

```diff
@@ -620,4 +623,68 @@ class AlignmentRecordRDDSuite extends ADAMFunSuite {
     assert(new File(outputPath1).exists())
     assert(new File(outputPath2).exists())
   }
+
+  sparkTest("don't lose any reads when piping as SAM") {
+    val reads12Path = resourcePath("reads12.sam")
+    val ardd = sc.loadBam(reads12Path)
+    val records = ardd.rdd.count
+
+    implicit val tFormatter = SAMInFormatter
+    implicit val uFormatter = new AnySAMOutFormatter
+
+    val pipedRdd: AlignmentRecordRDD = ardd.pipe("tee /dev/null")
+    val newRecords = pipedRdd.rdd.count
+    assert(records === newRecords)
+  }
+
+  sparkTest("don't lose any reads when piping as BAM") {
+    val reads12Path = resourcePath("reads12.sam")
+    val ardd = sc.loadBam(reads12Path)
+    val records = ardd.rdd.count
+
+    implicit val tFormatter = BAMInFormatter
+    implicit val uFormatter = new AnySAMOutFormatter
```

Comment:
Could you explain a bit how these implicits work?  And why the second one needs the `new`?  Do they have to be implicit?  Does them being implicit prevent calling the `pipe` method from Java?